enterprise/frontend/src/metabase-enterprise/whitelabel/index.js

```diff
@@ -69,6 +69,16 @@ if (hasPremiumFeature("whitelabel")) {
           display_name: t`Landing Page`,
           type: "string",
           placeholder: "/",
+          props: {
+            normalize(value, { type }) {
+              if (type === "text" && typeof value === "string") {
```

Comment:
```suggestion
              if (typeof value === "string") {
```
is it the same type defined a few lines above? If so we already know it's text in this case, and I think we should be fine with trimming all string values

Reply:
The reason is that there are more than just text inputs e.g. number input, but all of them will have [`value` as a string](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement#value). So I detect that from a `type` which tells what kind of input this will be instead.

![image](https://github.com/metabase/metabase/assets/1937582/86c48126-daf7-430d-b3cc-73085f0bbdf1)

And the value could also be `null`
https://github.com/metabase/metabase/blob/e21e425b011b134328cd26e794981bcfa18f568c/frontend/src/metabase/admin/settings/components/widgets/SettingInput.tsx#L7

so we need to check the type of the value as well before attempting to `trim()` it.

Reply:
I was referring to removing `type === "text"`, since we're passing this function directly to that specific widget, which is using a type="string" (at the setting level) it will use `type = "text"` as default (at the widget level).
That logic would make sense if we passed that function to all the settings by default,

Reply:
Ahh yeah, good point. This could all be simplified. Thanks for clarifying 🙏

Reply:
Fixed in https://github.com/metabase/metabase/pull/36693/commits/8a26f5420498ad4a8992ebcc21bb6465e3fe0b01